server/src/main/java/org/opensearch/gateway/remote/AbstractRemoteBlobStoreObject.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.gateway.remote;
+
+import org.opensearch.common.CheckedRunnable;
+import org.opensearch.common.blobstore.BlobPath;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.compress.Compressor;
+import org.opensearch.gateway.remote.ClusterMetadataManifest.UploadedMetadata;
+import org.opensearch.index.translog.transfer.BlobStoreTransferService;
+import org.opensearch.repositories.blobstore.BlobStoreRepository;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * An extension of {@link RemoteObject} class which caters to the use case of writing to and reading from a blob storage
+ *
+ * @param <T> The class type which can be uploaded to or downloaded from a blob storage.
+ */
+public abstract class AbstractRemoteBlobStoreObject<T> implements RemoteObject<T> {
+
+    public static final String PATH_DELIMITER = "/";
+
+    private final BlobStoreTransferService transferService;
+    private final BlobStoreRepository blobStoreRepository;
+    private final String clusterName;
+    private final ExecutorService executorService;
+
+    public AbstractRemoteBlobStoreObject(
+        BlobStoreTransferService blobStoreTransferService,
+        BlobStoreRepository blobStoreRepository,
+        String clusterName,
+        ThreadPool threadPool
+    ) {
+        this.transferService = blobStoreTransferService;
+        this.blobStoreRepository = blobStoreRepository;
+        this.clusterName = clusterName;
+        this.executorService = threadPool.executor(ThreadPool.Names.GENERIC);
+    }
+
+    public abstract BlobPathParameters getBlobPathParameters();
+
+    public abstract String getFullBlobName();
+
+    public String getBlobFileName() {
+        if (getFullBlobName() == null) {
+            generateBlobFileName();
+        }
+        String[] pathTokens = getFullBlobName().split(PATH_DELIMITER);
+        return getFullBlobName().split(PATH_DELIMITER)[pathTokens.length - 1];
```

Comment:
No need to split the Full Name again, we can use the pathTokens we have created in the above line.